### PR TITLE
DropDownList: it should be possible to set dropDownOptions.width smaller than editor width if dropDownOptions.minWidth is set by user (T916722)

### PR DESCRIPTION
**Pull request by ksercs**

### CODE REVIEW

**Review by DokaRus — DISMISSED** [review]

**Review by San4es — DISMISSED** [review]

File: js/ui/drop_down_editor/ui.drop_down_list.js [hunk 1]
```diff
@@ -744,16 +744,35 @@ const DropDownList = DropDownEditor.inherit({
         delete this._searchTimer;
     },
 
-    _updatePopupMinWidth() {
-        windowUtils.hasWindow() && this._popup && this._setPopupOption('minWidth', this.$element().outerWidth());
+    _updatePopupMinWidth(popupWidth) {
+        if(windowUtils.hasWindow() && this._popup) {
```
Comment (DokaRus) [inline js/ui/drop_down_editor/ui.drop_down_list.js]:
```suggestion
        if(window && this._popup) {
```
[L22](https://github.com/DevExpress/DevExtreme/blob/ed68457b6afcd0521d494707dd9bd3f0e5a7cbfe/js/ui/drop_down_editor/ui.drop_down_list.js#L22)

File: js/ui/drop_down_editor/ui.drop_down_list.js [hunk 2]
```diff
@@ -744,16 +744,35 @@ const DropDownList = DropDownEditor.inherit({
...
     _dimensionChanged: function() {
-        this._updatePopupMinWidth();
+        let popupWidth = this.option('dropDownOptions.width');
+        const popupMinWidth = this.option('dropDownOptions.minWidth');
+
+        if(popupWidth === null) {
+            popupWidth = undefined;
+        } else if(typeof popupWidth === 'function') {
+            popupWidth = popupWidth();
+        }
```
Comment (DokaRus) [inline js/ui/drop_down_editor/ui.drop_down_list.js]: Extract method?

File: js/ui/drop_down_editor/ui.drop_down_list.js (hunk 1 above)
Comment (ksercs) [reply js/ui/drop_down_editor/ui.drop_down_list.js]: fixed

File: js/ui/drop_down_editor/ui.drop_down_list.js (hunk 2 above)
Comment (ksercs) [reply js/ui/drop_down_editor/ui.drop_down_list.js]: fixed

**Review by DokaRus — APPROVED** [review]

**Pull request merged**